contracts/interchain-token-service/tests/utils/mod.rs

```diff
@@ -1,14 +1,56 @@
+use std::fmt::Debug;
+
 use axelar_gas_service::{AxelarGasService, AxelarGasServiceClient};
 use axelar_gateway::testutils::{setup_gateway, TestSignerSet};
 use axelar_gateway::AxelarGatewayClient;
+use axelar_soroban_std::events::Event;
+use axelar_soroban_std::impl_event_testutils;
 use axelar_soroban_std::types::Token;
 use interchain_token_service::{InterchainTokenService, InterchainTokenServiceClient};
-use soroban_sdk::{testutils::Address as _, token::StellarAssetClient, Address, Env, String};
+use soroban_sdk::Topics;
+use soroban_sdk::{
+    testutils::Address as _, token::StellarAssetClient, Address, Bytes, BytesN, Env, IntoVal,
+    String, Symbol, Val,
+};
 
 pub const HUB_CHAIN: &str = "hub_chain";
 
 const INTERCHAIN_TOKEN_WASM_HASH: &[u8] = include_bytes!("../testdata/interchain_token.wasm");
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct InterchainTransferReceivedEvent {
```

Comment:
Just wondering, is specifying `topic1, topic2, ...` necessary? It seems not, for example, OperatorshipTransferredEvent doesn't use it: https://github.com/axelarnetwork/axelar-cgp-soroban/blob/a90ac0b9d2d66533f32df8a9e5e15e483ec59c5d/packages/axelar-soroban-std/src/interfaces/operatable.rs#L48